pages/trips/[id]/index.js

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from "next/router.js";
+import useSWR from "swr";
+import Image from "next/image";
+import { formatDate } from "@/lib/utils";
+import PackingList from "@/components/PackingList";
+import NavigationButton from "@/components/NavigationButton";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: trip, isLoading, error } = useSWR(`/api/trips/${id}`);
+
+  if (!isReady || isLoading || error) return <h2>Loading...</h2>;
```

Comment:
If it's an error, the "Loading" screen may be misleading in some way. Because an error can lead to nothing loading.
You could think of something like this:
```suggestion
 if (error) return <h2>Error, please try again later...</h2>;
  if (!isReady || isLoading || error) return <h2>Loading...</h2>;
```